File: src/api/api_lists/models/item.py
"""
**********************************************************************************
List model
**********************************************************************************
"""
from enum import Enum
from dataclasses import dataclass
from uuid import UUID
from datetime import datetime



class ItemComplete(str, Enum):
    NO  = "n"
    YES = "y"
    

@dataclass
class Item:
    id          : UUID         = None         
    list_id     : UUID         = None         
    content     : str          = None         
    rank        : int          = None         
    complete    : ItemComplete = ItemComplete.NO
    created_on  : datetime     = None
    modified_on : datetime     = None

File: src/front-end/lists/static/js/custom/models/item.js

/**
 * Item model
 */
export class Item
{
    constructor(item) {
        this.complete    = null;
        this.content     = null;
        this.created_on  = null;
        this.id          = null;
        this.list_id     = null;
        this.list_name   = null;
        this.modified_on = null;
        this.rank        = null;

        for (const key in this) {
            if (item.hasOwnProperty(key)) {
                this[key] = item[key];
            }
        }
    }
}